astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
On the contrary, I think an array of `z` can be useful. For example, I have a spectrum at rest wavelength and I want to see what it would look like at several different redshifts for my research. I would rather evaluate all of them at once, since it is possible in `modeling`, rather than generate an instance for each `z`.